client/init.c

```diff
@@ -111,8 +111,12 @@ int xdag_init(int argc, char **argv, int isGui)
 		} else if(ARG_EQUAL(argv[i], "-i", "")) { /* interactive mode */
 			return terminal();
 		} else if(ARG_EQUAL(argv[i], "-z", "")) { /* memory map  */
-			if (++i < argc)
+			if (++i < argc) {
 				xdag_mem_tempfile_path(argv[i]);
+				if(!strcmp(argv[i], "RAM")) {
+					g_bi_index_enable = 0; /* disable bi index when use RAM */
+				}
```

Comment:
there is g_use_tmpfile variable that can be used instead of string comprae with "RAM", because for example in windows g_use_tmpfile is disabled too even without -z RAM.

the best place to set g_bi_index_enable is

```
int xdag_blocks_start(int is_pool, int mining_threads_count, int miner_address)
{
	pthread_mutexattr_t attr;
	pthread_t th;

	if (!is_pool) {
		g_light_mode = 1;
	}

	if (xdag_mem_init(g_light_mode && !miner_address ? 0 : (((get_timestamp() - XDAG_ERA) >> 10) + (uint64_t)365 * 24 * 60 * 60) * 2 * sizeof(struct block_internal))) {
		return -1;
	}
<-------------------- ***here*** ----- you can check for g_use_tmpfile and set g_bi_index_enable
	pthread_mutexattr_init(&attr);
	pthread_mutexattr_settype(&attr, PTHREAD_MUTEX_RECURSIVE);
	pthread_mutex_init(&block_mutex, &attr);
	pthread_mutex_init(&rbtree_mutex, 0);
	int err = pthread_create(&th, 0, work_thread, (void*)(uintptr_t)(unsigned)mining_threads_count);
```